src/main/src/main/java/org/geoserver/catalog/impl/DefaultCatalogFacade.java

```diff
@@ -200,10 +200,17 @@ public void remove(StoreInfo store) {
     }
     
     public void save(StoreInfo store) {
-        beforeSaved(store);
+        ModificationProxy h = (ModificationProxy) Proxy.getInvocationHandler(store);
```

Comment:
That sounds like a good idea, although I don't know how to implement it (I am still not especially familiar with lambdas in java). If you've got an example of what you are suggesting, I could certainly apply it to the relevant places.